### DISCUSSION TITLE
9.5.2 dev server does not respond to requests

### DISCUSSION THREAD

**Discussion by tomaszrondio** [vercel/next.js#16302] [Help]
# Bug report

## Describe the bug

Server is unresponsive in dev mode. Requests are pending forever and get cancelled. In "production" mode it works fine.
Updated from 9.4.6 
I use next-pwa

## Screenshots

<img width="847" alt="Screenshot 2020-08-18 at 12 17 19" src="https://user-images.githubusercontent.com/4939724/90502245-cc078e00-e14d-11ea-9b5c-50196ff3b027.png">

## System information

- OS: macOS
- Browser chrome
- Version of Next.js:  9.5.2
- Version of Node.js: 10.15.0

**Comment by daveteu** [comment]
did u use asPath in your code? #16269

**Comment by tomaszrondio** [comment]
I do to check if menu item should be active or not.
```
  const router = useRouter();
  const indexOfQuestionMark = router.asPath.indexOf("?");
  const asPath = router.asPath.substring(
    0,
    indexOfQuestionMark !== -1 ? indexOfQuestionMark : router.asPath.length
  );
  const linkClassName = asPath === href ? "active" : "";
```

All Links are generated like this: 
`<Link href="/[...slug]" as={href}>` 
where `href` is a string from database. `[...slug]` catches all routes except homepage

**Comment by tomaszrondio** [comment]
funny thing is that sometimes it works and at some point it stops to work. Then if I restart the server it works again. There must be some bug that does not throw an error message.

**Comment by tomaszrondio** [comment]
I disabled next-pwa for dev environment and it looks that it helped.

**Comment by timneutkens** [comment]
Converting to a discussion as there is no reproduction so it's impossible to investigate.

**Reply by tomaszrondio** [reply]
I can provide whole repository. I also have issues with server side rendering in dev mode. Like it ignores getStaticProps and renders page without data. It looks that my setup is broken. I reinstalled dependencies but it didn't help. Need to figure it out or burn it :)

**Comment by tomaszrondio** [comment]
@timneutkens 
ok so the thing is:
if U put this code into `about.jsx` file in pages folder and go to localhost:3000/about it will render on server side properly.
But if you put it into `[...slug].jsx` it won't. It will render component with empty props. Basically it does not wait for getStaticProps on server side. In the end it will render with props but on client side only.

```
import React from "react";

const Pages = (props) => {
  return <>{props.template}</>;
};

export async function getStaticProps(context) {
  return {
    props: { template: "SomeRandomValue" },
    revalidate: 1,
  };
}

export async function getStaticPaths() {
  return {
    paths: [],
    fallback: true,
  };
}

export default Pages;
```

**Reply by tomaszrondio** [reply]
this is how part of response looks like:
```
<script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{},"__N_SSG":true},"page":"/[...slug]","query":{"slug":["local"]},"buildId":"development","nextExport":false,"isFallback":true,"gsp":true}</script>
<script src="/_next/static/chunks/pages/_app.js?ts=1597920184797"></script>
<script src="/_next/static/chunks/pages/%5B...slug%5D.js?ts=1597920184797">
```
pageProps are empty

**Reply by timneutkens** [reply]
This is expected behavior as per the fallback: true documentation: https://nextjs.org/docs/basic-features/data-fetching#fallback-true, fallback means you render without props on the very first request and then it's generated in the background

**Reply by tomaszrondio** [reply]
oh man... I missed that in documentation. Now I get 404 which is an expected behaviour but kinda useless if [..slug] does not catch all routes anymore. If I understand correctly it is actually good that I set fallback to true. But it makes it impossible to develop AMP version of pages.

**Reply by tomaszrondio** [reply]
p.s. what is interesting as well is that it works for defined routes but not for "catch all routes".

**Reply by tomaszrondio** [reply]
background: I am working with CMS. All routes are dynamic. You never know what kind of content will be under a route. I have `index.jsx` for main page and `[..slug].jsx` for all other pages. Inside `getStaticProps` in `[..slug].jsx` I call the database to get kind of route and data for that route. It can be a section or article. From there I render proper component that displays the data. As I want to have AMP version of articles I need to know that given route is an article on server side.

**Reply by timneutkens** [reply]
Something that's still in development but you can try it out is `fallback: 'unstable_blocking'` which makes the first request a blocking server-side render which would allow you to ship AMP

**Reply by tomaszrondio** [reply]
great! That works as I need it.
